src/pip/_internal/cli/cmdoptions.py

```diff
@@ -335,6 +335,17 @@ def exists_action():
 )  # type: Callable[..., Option]
 
 
+def extra_headers():
```

Comment:
You need to add it to the `install` and `download` (and to probably more ones that I don't remember) CLI. You should add it as a req file option

Reply:
thanks for the hints, I wasn't quite sure where to drop this code.

Reply:
Would it be okay to put it in the `general_group`?

Reply:
Forgive my ignorance, but what do you mean "You should add it as a req file option"?

Reply:
For the req file- see SUPPORTED_VERSIONS in req_file.py

About the general- it does not makes sense as there are pip commands which do not make any requests at all, you can add to install cmd, download cmd, etc...